app/javascript/src/controller_branches.js

```diff
@@ -78,6 +78,11 @@ BranchesController.enhanceCurrentBranches = function($branches) {
     if(index == 0) {
       branch.$node.find(".BranchRemover").eq(0).hide();
     }
+    
+    // Inject or statements between each of the branches 
+    if(index != 0) {
+      branch.$node.before("<p class=\"branch-or\">or</p>");
```

Comment:
Looks like inspiration for this approach might have come from the view-specific BranchInjector class, which also has a line `$node.before("<p class=\"branch-or\">or</p>")` featuring in an AJAX response handler. 

Can't say I'm convinced the earlier approach is right and, if it was the inspiration, then it shows how a small error can easily become bigger. With this new addition, we now have two places inside JS code that are injecting DOM elements that could/should be handled from a template. 

I'm wondering if adjusting view templates to include the branch-or element would be the fix required to remove both these JS-based DOM injections.

Reply:
I had a feeling you might say soemthing along these lines 😂 I mostly just continued with the approach as I was treating it as a quick bugfix rather than a refactor, and didn't necessarily want to break stuff in the process.  But I'm inclined to agree that this approach isn't necessarily the best.  I'll revisit it and attempt to improve.